File: modules/angular2/test/reflection/reflector_common.dart
class ClassDecorator {
  final dynamic value;

  const ClassDecorator(this.value);
}

class ParamDecorator {
  final dynamic value;

  const ParamDecorator(this.value);
}

ClassDecorator classDecorator(value) {
  return new ClassDecorator(value);
}

ParamDecorator paramDecorator(value) {
  return new ParamDecorator(value);
}

File: modules/angular2/test/core/reflection/reflector_common.ts
import {
  makeDecorator,
  makeParamDecorator,
  makePropDecorator
} from 'angular2/src/core/util/decorators';

export class ClassDecoratorMeta {
  constructor(public value) {}
}

export class ParamDecoratorMeta {
  constructor(public value) {}
}

export class PropDecoratorMeta {
  constructor(public value) {}
}

export function classDecorator(value) {
  return new ClassDecoratorMeta(value);
}

export function paramDecorator(value) {
  return new ParamDecoratorMeta(value);
}

export function propDecorator(value) {
  return new PropDecoratorMeta(value);
}

export var ClassDecorator = makeDecorator(ClassDecoratorMeta);
export var ParamDecorator = makeParamDecorator(ParamDecoratorMeta);
export var PropDecorator = makePropDecorator(PropDecoratorMeta);
